lib/peer.js

```diff
@@ -173,13 +184,72 @@ class Peer {
       this.destroy(err)
     })
 
+    wire.once('pe1', () => {
+      this.onPe1()
+    })
+    wire.once('pe2', () => {
+      this.onPe2()
+    })
+    wire.once('pe3', () => {
+      this.onPe3()
+    })
+    wire.once('pe4', () => {
+      this.onPe4()
+    })
     wire.once('handshake', (infoHash, peerId) => {
       this.onHandshake(infoHash, peerId)
     })
     this.startHandshakeTimeout()
 
     conn.pipe(wire).pipe(conn)
-    if (this.swarm && !this.sentHandshake) this.handshake()
+    if (this.swarm) {
+      if (this.type === 'tcpOutgoing') {
+        if (this.retries === 0 && !this.sentPe1) this.sendPe1()
```

Comment:
shouldn't this be flagged with `secured`?

Reply:
added